File: examples-sdk/go/022-basic-sidecar-python/main.go
// 022-basic-sidecar-python — Go SDK equivalent
// Compile: vil compile --from go --input 022-basic-sidecar-python/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	s := vil.NewServer("sidecar-python-example", 8080)

	fraud := vil.NewService("fraud")
	fraud.Endpoint("GET", "/status", "fraud_status")
	fraud.Endpoint("POST", "/check", "fraud_check")
	s.Service(fraud)

	root := vil.NewService("root")
	root.Endpoint("GET", "/", "index")
	s.Service(root)

	s.Compile()
}

File: examples-sdk/zig/022-basic-sidecar-python/main.zig
// 022-basic-sidecar-python — Zig SDK equivalent
// Compile: vil compile --from zig --input 022-basic-sidecar-python/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var server = vil.Server.init("sidecar-python-example", 8080);
    var fraud = vil.Service.init("fraud");
    fraud.endpoint("GET", "/status", "fraud_status");
    fraud.endpoint("POST", "/check", "fraud_check");
    server.service(&fraud);
    var root = vil.Service.init("root");
    root.endpoint("GET", "/", "index");
    server.service(&root);
    server.compile();
}
